pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -293,12 +331,41 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 	}
 
 	reconciler.EmitEvent(c.Recorder, before, after, tr)
-
 	c.Logger.Infof("Successfully reconciled taskrun %s/%s with status: %#v", tr.Name, tr.Namespace, after)
 
 	return nil
 }
 
+func (c *Reconciler) sendCloudEvents(tr *v1alpha1.TaskRun) error {
```

Comment:
i have a feeling im repeating myself in which case i apologize but i really would like to avoid adding functions to the reconciler as much as possible - id rather see this as a function in a separate package called by the reconcile function. it seems like the reconciler is providing only a logger and a client which should be pretty easy to pass in instead of giving the function full access to the reconciler

Reply:
done